geom_mgr/CMakeLists.txt

```diff
@@ -26,15 +43,41 @@ set(srcs
   latlon/LatLonGeomFactory_smod.F90
 
   GeomManager.F90
-  GeomManager_smod.F90
+  GeomManager/new_GeomManager_GeomManager.F90
+  GeomManager/initialize_GeomManager.F90
+  GeomManager/add_factory_GeomManager.F90
+  GeomManager/delete_mapl_geom_GeomManager.F90
+  GeomManager/get_mapl_geom_from_hconfig_GeomManager.F90
+  GeomManager/get_mapl_geom_from_metadata_GeomManager.F90
+  GeomManager/get_mapl_geom_from_id_GeomManager.F90
+  GeomManager/get_mapl_geom_from_spec_GeomManager.F90
+  GeomManager/add_mapl_geom_GeomManager.F90
+  GeomManager/make_geom_spec_from_metadata_GeomManager.F90
+  GeomManager/make_geom_spec_from_hconfig_GeomManager.F90
+  GeomManager/make_mapl_geom_from_spec_GeomManager.F90
+  GeomManager/get_geom_from_id_GeomManager.F90
+  #GeomManager_smod.F90
 
 # gFTL containers
   GeomFactoryVector.F90
   GeomSpecVector.F90
   IntegerMaplGeomMap.F90
 
   VectorBasis.F90
-  VectorBasis_smod.F90
+  VectorBasis/create_fields_VectorBasis.F90
+  VectorBasis/destroy_fields_VectorBasis.F90
+  VectorBasis/get_unit_vector_VectorBasis.F90
+  VectorBasis/grid_get_centers_VectorBasis.F90
+  VectorBasis/grid_get_coords_1d_VectorBasis.F90
+  VectorBasis/grid_get_coords_2d_VectorBasis.F90
+  VectorBasis/grid_get_corners_VectorBasis.F90
+  VectorBasis/latlon2xyz_VectorBasis.F90
+  VectorBasis/MAPL_GeomGetCoords_VectorBasis.F90
+  VectorBasis/mid_pt_sphere_VectorBasis.F90
+  VectorBasis/new_GridVectorBasis_VectorBasis.F90
+  VectorBasis/new_NS_Basis_VectorBasis.F90
+  VectorBasis/xyz2latlon_VectorBasis.F90
```

Comment:
OK - this get a bit bloated in the CMakeLists.   We can instead add a very simple CMakeLists.txt in the subdirectory.  Lok at how I do this for `generic3g/specs`.